src/commands/support/help.js

```diff
@@ -1,198 +1,78 @@
-
-	else if (args[0] == 'moderation') {
-		return message.channel.send({ embeds: [moderation] });
-	}
-	else if (args[0] == 'fun') {
-		return message.channel.send({ embeds: [fun] });
-	}
-	else if (args[0] == 'actions') {
-		return message.channel.send({ embeds: [actions] });
-	}
-	else if (args[0] == 'gifs') {
-		return message.channel.send({ embeds: [gifs] });
-	}
-	else if (args[0] == 'support') {
-		return message.channel.send({ embeds: [support] });
-	}
-	else if (args[0] == 'dev') {
-		return message.channel.send({ embeds: [dev] });
-	}
-	else if (args[0] == 'images') {
-		return message.channel.send({ embeds: [images] });
-	}
-	else if (args[0] == 'utils') {
-		return message.channel.send({ embeds: [utils] });
-	}
-	else if (args[0] == 'about') {
-		return message.channel.send({ embeds: [about] });
-	}
-	else if (args[0] == 'media') {
-		return message.channel.send({ embeds: [media] });
-	}
-	else if (args[0] == 'slashcommands') {
-		return message.channel.send({ embeds: [slashcommands] });
-	}
-	else {
-		return message.channel.send({ embeds: [help] });
-	}
+const { MessageEmbed, MessageButton, MessageActionRow } = require('discord.js');
+const { readdirSync } = require('fs');
+
+module.exports = {
+	name: 'help',
+	description: 'Provides a list of all my commands!',
+	usage: '[command]',
+
+	permissions: [],
+	ownerOnly: false,
+	guildOnly: false,
+
+	options: [
+		{ name: 'command', description: 'Shows details about how to use a command', type: 'STRING', required: false },
+	],
+
+	error: false,
+	execute: async ({ interaction, client }) => {
+
+		const cmdName = interaction.options.getString('command');
+		const cmd = client.commands.get(cmdName);
+
+		if (cmd) {
+
+			const embed = new MessageEmbed()
+				.setColor('#0099FF')
+				.setTitle(cmd.name.charAt(0).toUpperCase() + cmd.name.slice(1) + ' Command')
+				.setURL('https://dsc.gg/kekinv')
+				.setDescription(cmd.description)
+				.setTimestamp();
+
+			embed.addField('__Usage:__', '/' + cmd.name + (cmd.usage ? ' ' + cmd.usage : ''), false);
+
+			if (cmd.permissions[0] && cmd.ownerOnly == false) {
+				embed.addField('__Permissions:__', '`' + cmd.permissions.join('` `') + '`', false);
+			}
+			if (!cmd.permissions[0] && cmd.ownerOnly == true) {
+				embed.addField('__Permissions:__', '**Server Owner Only**', false);
+			}
+			if (cmd.error == true) {
+				embed.addField('__Error:__', 'This command is currently unavailable, please try again later.', false);
+			}
+
+			interaction.followUp({ embeds: [embed], ephemeral: false });
+
+		}
+		else {
+
+			const embed = new MessageEmbed()
+				.setColor('#0099FF')
+				.setTitle(client.user.username + ' Commands')
+				.setURL('https://dsc.gg/kekinv')
+				.setDescription('To view the information about a certain command\ndo `/help <command>`.')
+				.setThumbnail(client.user.displayAvatarURL())
+				.setTimestamp();
+
+			for (const category of ['fun', 'moderation', 'support']) {
```

Comment:
Change the array to include all command directories - `botinfo`, `fun`, `media`, `moderation` and `support` - in order to include all commands in the name help page.